src/MetaMorpho.sol

```diff
@@ -375,7 +375,7 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
             uint256 withdrawn = supplyAssets.zeroFloorSub(allocation.assets);
 
             if (withdrawn > 0) {
-                if (allocation.marketParams.loanToken != asset()) revert ErrorsLib.InconsistentAsset(id);
+                if (!config[id].enabled) revert ErrorsLib.MarketNotEnabled(id);
```

Comment:
Should be possible to withdraw from a market that has the pending removal on timelock?

Reply:
Yes, users should be able to exit before it's removed. It's required to keep the vault noncustodial.

Reply:
Is this comment valid also for the `reallocate` function?

Reply:
Yes it is, for reallocate the goal is to retrieve funds from this market before it's removed